packages/filesystem/src/node/nsfw-watcher/nsfw-filesystem-watcher.ts

```diff
@@ -130,30 +129,39 @@ export class NsfwFileSystemWatcherServer implements FileSystemWatcherServer {
                     console.warn(`Failed to watch "${basePath}":`, error);
                     this.unwatchFileChanges(watcherId);
                 }
+            }).then((watcher: nsfw.NSFW | undefined) => {
+                if (!watcher) {
```

Comment:
how can it be `undefined` here?

Reply:
It cannot be, but based on the original code, we set the reference to `undefined`, so this is the only way to trick the compiler. By the way, see the [original code](https://github.com/theia-ide/theia/pull/5111/files#diff-8053c51c6deb015c4dcc93ed38f9daa9L111), it was `undefined` there as well. I think this change is correct.

Reply:
i see, it seems before tsc was smart enough to figure out that it cannot be undefined after calling `nsfw`